components/org.wso2.bfsi.identity.extensions/src/main/java/org/wso2/bfsi/identity/extensions/auth/extensions/response/validator/BFSIHybridResponseTypeValidator.java

```diff
@@ -0,0 +1,86 @@
+ * <p>
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.bfsi.identity.extensions.auth.extensions.response.validator;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.oltu.oauth2.as.validator.TokenValidator;
+import org.apache.oltu.oauth2.common.OAuth;
+import org.apache.oltu.oauth2.common.error.OAuthError;
+import org.apache.oltu.oauth2.common.exception.OAuthProblemException;
+import org.wso2.carbon.identity.oauth.common.OAuthConstants;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+   Validator for hybrid flow code token requests.
+ */
+public class BFSIHybridResponseTypeValidator extends TokenValidator {
+
+    private static final Log log = LogFactory.getLog(BFSIHybridResponseTypeValidator.class);
+
+    private static boolean isContainOIDCScope(String scope) {
+
+        String[] scopeArray = scope.split("\\s+");
+        for (String anyScope : scopeArray) {
+            if (anyScope.equals(OAuthConstants.Scope.OPENID)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    @SuppressFBWarnings("SERVLET_PARAMETER")
+    // Suppressed content - request.getParameter("response_type")
+    // Suppression reason - False Positive : These endpoints are secured with access control
+    // as defined in the IS deployment.toml file
+    // Suppressed warning count - 3
+    public void validateRequiredParameters(HttpServletRequest request) throws OAuthProblemException {
+
+        super.validateRequiredParameters(request);
+        String openIdScope = request.getParameter(OAuth.OAUTH_SCOPE);
+
+        if (StringUtils.isBlank(openIdScope) || !isContainOIDCScope(openIdScope)) {
+            String clientID = request.getParameter(OAuth.OAUTH_CLIENT_ID).replaceAll("[\r\n]", "");
+            throw OAuthProblemException.error(OAuthError.CodeResponse.INVALID_REQUEST)
+                    .description("Request with \'client_id\' = \'" + clientID +
+                            "\' has \'response_type\' for \'hybrid flow\'; but \'openid\' scope not found.");
+        }
+
+    }
+
+    @Override
+    public void validateMethod(HttpServletRequest request) throws OAuthProblemException {
+
+        String method = request.getMethod();
+        if (!OAuth.HttpMethod.GET.equals(method) && !OAuth.HttpMethod.POST.equals(method)) {
+            throw OAuthProblemException.error(OAuthError.CodeResponse.INVALID_REQUEST)
+                    .description("Method not correct.");
```

Comment:
```suggestion
                    .description("HTTP method not correct.");
```

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/6/commits/884f24f4261bc92a886a68a4e8a5bf56e7a61136